### PR TITLE
[JB][OPS-9801][OPS-9802]

### PR DESCRIPTION
**Pull request by barnesjake**
- sign out functionality
- timeout functionality

### CODE REVIEW

File: app/uk/gov/hmrc/directdebitupdateemailfrontend/controllers/SignOutController.scala [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+) extends FrontendController(mcc) {
+
+  def signOutFromTimeout: Action[AnyContent] = Action { implicit request =>
+    // N.B. the implicit request being passed into the page here may still have the auth
+    // token in it so take care to ensure that the sign out link is not shown by mistake
+    Ok(timedOutPage()).withNewSession
+  }
+
+  def signOut: Action[AnyContent] = actions.authenticatedJourneyAction { _ =>
+    Redirect(appConfig.BaseUrl.govUkUrl)
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/directdebitupdateemailfrontend/controllers/SignOutController.scala:42]: do we not need to do `.withNewSession` here to actually log out?

File: app/uk/gov/hmrc/directdebitupdateemailfrontend/views/Layout.scala.html [hunk 2]
```diff
@@ -33,17 +35,35 @@
...
+        appConfig: AppConfig
 )
-@(pageTitle: String,
+@(
+        pageTitle: String,
         hasErrors: Boolean = false,
         showBackLink: Boolean = true,
-        backLinkUrlOverride: Option[String] = None
+        backLinkUrlOverride: Option[String] = None,
+        showSignOutOverride: Option[Boolean] = None
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/directdebitupdateemailfrontend/views/Layout.scala.html:47]: do we need this override? Is it used anywhere?

File: app/uk/gov/hmrc/directdebitupdateemailfrontend/controllers/SignOutController.scala (hunk 1 above)
Comment (AndyHWChung) [reply app/uk/gov/hmrc/directdebitupdateemailfrontend/controllers/SignOutController.scala:42]: also, is this actually being used? It looks like the signing out behaviour is pointing to the url you read from config

File: app/uk/gov/hmrc/directdebitupdateemailfrontend/views/Layout.scala.html (hunk 2 above)
Comment (barnesjake) [reply app/uk/gov/hmrc/directdebitupdateemailfrontend/views/Layout.scala.html:47]: Probably not, put it in incase, but I'll remove

File: app/uk/gov/hmrc/directdebitupdateemailfrontend/controllers/SignOutController.scala (hunk 1 above)
Comment (barnesjake) [reply app/uk/gov/hmrc/directdebitupdateemailfrontend/controllers/SignOutController.scala:42]: Oh yeah duh, will update the signout url in the template

File: app/uk/gov/hmrc/directdebitupdateemailfrontend/views/Layout.scala.html (hunk 2 above)
Comment (barnesjake) [reply app/uk/gov/hmrc/directdebitupdateemailfrontend/views/Layout.scala.html]: done

File: app/uk/gov/hmrc/directdebitupdateemailfrontend/controllers/SignOutController.scala (hunk 1 above)
Comment (AndyHWChung) [reply app/uk/gov/hmrc/directdebitupdateemailfrontend/controllers/SignOutController.scala:42]: we probably don't need this still if we make both the timeout and the actual sign out button point to the url in config. It looks like we can't control the continue url with the auth login stub though - but we can for the real thing 

https://github.com/hmrc/bas-gateway-frontend/blob/main/conf/app.routes#L14

we should also raise a PR for app-config-prod to make sure we've got the right sign-out url there. We'd need proper test data to be able to sign in and test it in QA though 🤔

Comment (barnesjake) [reply app/uk/gov/hmrc/directdebitupdateemailfrontend/controllers/SignOutController.scala:42]: Yeah I was going to raise some prs for app-config-<envs>. 
I'll remove the signout action from the controller now

File: conf/application.conf [hunk 3]
```diff
@@ -183,6 +183,8 @@ baseUrl {
   accessibility-statement-frontend-local = "http://localhost:12346"
   direct-debit-update-email-frontend-local = "http://localhost:10801"
   gg = "http://localhost:9949/auth-login-stub/gg-sign-in"
+  sign-out = "http://localhost:9949/auth-login-stub/session/logout"
+  govUk = "https://www.gov.uk"
```
Comment (AndyHWChung) [inline conf/application.conf:187]: I don't think this is being used anymore - this could be removed

**Review by AndyHWChung — APPROVED** [review]

**Pull request merged**